syntax = "proto3";

import "davStreamletSetting.proto";
import "appGlobalSetting.proto";

package IalConfig;

message IalGlobalSetting {
    int32 max_participants = 10; /* should less than 25 */
}

message IalTaskConfig {
    AppGlobalSetting.GlobalSetting app_global_setting = 1;
    IalGlobalSetting ial_global_setting = 2;
    DavStreamletSetting.InputStreamletSetting input_setting = 10;
    DavStreamletSetting.MixStreamletSetting mix_setting = 11;
    /* an unique tag associated with one set of output settting */
    map<string, DavStreamletSetting.OutputStreamletSetting> output_settings = 12;
}